daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/ScenarioRunner.scala

```diff
@@ -29,6 +29,7 @@ final case class ScenarioRunner(
     machine: Speedy.Machine,
     partyNameMangler: (String => String) = identity) {
   var ledger: ScenarioLedger = ScenarioLedger.initialLedger(Time.Timestamp.Epoch)
+  val onLedger = machine.withOnLedger("ScenarioRunner")(identity)
```

Comment:
This feels a bit hacky. Maybe that's a bias coming from Haskell, but to me `withOnLedger` implies that the `onLedger` should be limited to the scope of the continuation and should not leak out. Either `machine.getOnLedger` or moving `machine.withOnLedger` to the relevant places would seem cleaner.

Reply:
Yeah I agree that it’s a bit weird. The reason why I went for this is that the machine is t the class level so it seems weird to limit the scope of the onledger stuff to anything less.

Reply:
I’ve inlined the implementation in those two cases to make this a bit less confusing.

Reply:
Thanks, yes, that's clearer.